wled00/FX.cpp

```diff
@@ -1212,19 +1212,21 @@ uint16_t mode_fireworks() {
 
   bool valid1 = (SEGENV.aux0 < width*height);
   bool valid2 = (SEGENV.aux1 < width*height);
+  uint8_t x = SEGENV.aux0%width, y = SEGENV.aux0/width; // 2D coordinates stored in upper and lower byte
```

Comment:
It might be better to use `uint16_t` here, to also support "extreme" matrix sizes like 300 x 3.
Also uint8_t is very slow, as each computation includes an explicit " & 0xFF" to enforce correct overflows.

Reply:
Matrices with dimensions greater than 255x255 are not supported ATM.
They would require change to Segment and Panel classes.

I do agree that forcibly constraining a counter may not be the best idea. Perhaps we should change every unsigned counter/variable to use `size_t` instead, which should use optimised word size (usually `unsigned int`).

Reply:
> I do agree that forcibly constraining a counter may not be the best idea. Perhaps we should change every unsigned counter/variable to use `size_t` instead, which should use optimised word size (usually `unsigned int`).

Hi, thanks for the clarification.
This "loop/counter" variable thing is kind of personal preference, as long as its efficient. My point was just that `uint8_t` is not efficient (w.r.t speed) on esp32, as esp32 emulates 8bit with 16bit registers, and you might even have penalties for misaligned addresses of an 8bit value.

(slightly off-topic)
I must admit I'm not a friend of `size_t` when used as a generic type, because the standards define it as "type of the result of the sizeof operator". Its a special purpose type, like `time_t` is the type of unix epoch time.

I would suggest this for loops and counters:
* use `int` or `unsigned` when the results will fit into 16 bit. This type is guaranteed to be a "native" register size on any architecture (esp32, 8266, or -C3). Additionally, `int` and `unsigned` are always stored with native alignment.
* use `fast` types when more bytes are needed, and the compiler will pick the most efficient implementation for us.

basic types: https://en.cppreference.com/w/cpp/language/types
fixed-width types: https://en.cppreference.com/w/cpp/types/integer

Reply:
I agree with every point.
I did start to use `int` (mostly in `for` loops) but was not thorough enough.